plugins/jira/commands/issues-by-component.md

```diff
@@ -0,0 +1,674 @@
+  API_URL="${JIRA_URL}/rest/api/2/search?\
+   jql=${ENCODED_JQL}&\
+   startAt=${START_AT}&\
+   maxResults=1000&\
+   fields=summary,status,priority,assignee,reporter,created,updated,description,labels,components,issuetype"
+
+  # Fetch batch using curl with Bearer token authentication
+  HTTP_CODE=$(curl -s -w "%{http_code}" \
+    -o ".work/jira-issues-by-component/${PROJECT_KEY}/batch-${BATCH_NUM}.json" \
+    -H "Authorization: Bearer ${AUTH_TOKEN}" \
+    -H "Accept: application/json" \
+    "${API_URL}")
+
+  # Check HTTP response code
+  if [ "$HTTP_CODE" -ne 200 ]; then
+    echo "Error: HTTP $HTTP_CODE received"
+    cat ".work/jira-issues-by-component/${PROJECT_KEY}/batch-${BATCH_NUM}.json"
+    exit 1
+  fi
+
+  # Parse response to check if more results exist
+  BATCH_SIZE=$(jq '.issues | length' ".work/jira-issues-by-component/${PROJECT_KEY}/batch-${BATCH_NUM}.json")
+  TOTAL=$(jq '.total' ".work/jira-issues-by-component/${PROJECT_KEY}/batch-${BATCH_NUM}.json")
+
+  TOTAL_FETCHED=$((TOTAL_FETCHED + BATCH_SIZE))
+
+  echo "✓ Fetched ${BATCH_SIZE} issues (${TOTAL_FETCHED}/${TOTAL} total)"
+
+  # Check if done
+  if [ ${TOTAL_FETCHED} -ge ${TOTAL} ] || [ ${BATCH_SIZE} -eq 0 ]; then
+    break
+  fi
+
+  # Move to next batch
+  START_AT=$((START_AT + 1000))
+  BATCH_NUM=$((BATCH_NUM + 1))
+done
+
+echo ""
+echo "✓ Fetching complete: ${TOTAL_FETCHED} issues downloaded in $((BATCH_NUM + 1)) batch(es)"
+```
+
+**Why curl instead of MCP:**
+- Direct file streaming avoids Claude's tool result size limits (413 errors)
+- Can handle thousands of tickets without token consumption
+- Faster - no intermediate serialization through MCP protocol
+- More reliable for large datasets
+
+### 5. Process Batches with Python to Group by Component and Apply Search
+
+Create a Python script (`.work/jira-issues-by-component/{project-key}/process_batches.py`) that:
+
+**Inputs:**
+- All batch files: `.work/jira-issues-by-component/{project-key}/batch-*.json`
+- Search parameters from command-line arguments
+
+**Processing Logic:**
+```python
+import json
+import glob
+import sys
+from collections import defaultdict
+from datetime import datetime
+
+# Parse command-line arguments
+search_terms = sys.argv[1] if len(sys.argv) > 1 else None
+search_description = sys.argv[2] == "true" if len(sys.argv) > 2 else False
+
+# Load all batches
+all_issues = []
+for batch_file in sorted(glob.glob('.work/jira-issues-by-component/{project-key}/batch-*.json')):
+    with open(batch_file) as f:
+        data = json.load(f)
+        all_issues.extend(data['issues'])
+
+# Apply text search filter if provided
+if search_terms:
+    keywords = [k.strip().lower() for k in search_terms.replace(',', ' ').split()]
+    filtered_issues = []
+
+    for issue in all_issues:
+        summary = issue['fields'].get('summary', '').lower()
+        description = issue['fields'].get('description', '').lower() if search_description else ''
+
+        # Check if any keyword matches
```

Comment:
I think we don’t need any Python code, we should use the JIRA MCP.

Reply:
@LiangquanLi930 Thanks for reviewing. Makes sense for removing python code reference. Use of API versus MCP is transparent to user.

Let me know if the commit needs to be squashed.